contracts/v0.8/utils/Errors.sol

```diff
@@ -0,0 +1,40 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.18;
+
+library Errors {
+    error NonZeroExitCode(int256 exit_code, string description);
+
+    // Exit codes
+    int256 constant USR_ILLEGAL_ARGUMENT = 16;
+    int256 constant USR_NOT_FOUND = 17;
+    int256 constant USR_FORBIDDEN = 18;
+    int256 constant USR_INSUFFICIENT_FUNDS = 19;
+    int256 constant USR_ILLEGAL_STATE = 20;
+    int256 constant USR_SERIALIZATION = 21;
+    int256 constant USR_UNHANDLED_MESSAGE = 22;
+    int256 constant USR_UNSPECIFIED = 23;
+    int256 constant USR_ASSERTION_FAILED = 24;
+    int256 constant START_FOR_ACTOR_SPECIFIC_EXIT_CODES = 32;
+
+    function revertWithDescriptionIfExitCodeIsNotZero(int256 exit_code) internal pure {
```

Comment:
This name is a bit of a mouthful, I'd rename to something like `revertOnError()` (it can still create the correct description, it would just be an implementation detail.